llvm/lib/Transforms/Yk/BasicBlockTracer.cpp

```diff
@@ -0,0 +1,72 @@
+#include "llvm/Transforms/Yk/BasicBlockTracer.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+#include "llvm/InitializePasses.h"
+#include "llvm/Pass.h"
+
+#define DEBUG_TYPE "yk-basicblock-tracer-pass"
+
+using namespace llvm;
+
+namespace llvm {
+void initializeYkBasicBlockTracerPass(PassRegistry &);
+} // namespace llvm
+
+namespace {
+struct YkBasicBlockTracer : public ModulePass {
```

Comment:
I wonder if a name like `YkSoftwareTracer` would be more indicative of what this does?

(an annoying change to request, I know)

Reply:
I had thought that too originally, but I think we *could* use this for other purposes (at least in the simple sense that the pass is currently implemented), so the generic name seems OK for now. If the pass becomes more specific, then I think we can make the name more specific later.